scipy/stats/_stats_mstats_common.py

```diff
@@ -298,7 +310,10 @@ def theilslopes(y, x=None, alpha=0.95):
     slopes = deltay[deltax > 0] / deltax[deltax > 0]
     slopes.sort()
     medslope = np.median(slopes)
-    medinter = np.median(y) - medslope * np.median(x)
+    if method == 'joint':
+        medinter = np.median(y - medslope * x)
+    else:
```

Comment:
write `elif method == 'separate': ... else raise ValueError('...')` or check that method is valid at the beginning of the function